src/gt4py/storage/storage.py

```diff
@@ -207,6 +207,11 @@ def __array_finalize__(self, obj):
                     raise RuntimeError(
                         "Meta information can not be inferred when creating Storage views from other classes than Storage."
                     )
+                if self.ndim != obj.ndim and self.ndim != 0:
```

Comment:
What case does the self.ndim != 0 cover? So it's still allowed to access scalar elements?

Reply:
This is not 100% clean, but I would say the benefits of allowing scalar storages (with incorrect metadata) overweight here. For example `numpy.allclose` relies on reducing an array/storage of booleans to a scalar boolean, so disallowing this case would mean require all occurences.